File: Data Structures & Algorithms/maximum-product-subarray/submission-3.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n= nums.size();
        if(n==1) return nums[0];
        int res =0 ;
        for(int i=0;i<n;i++){
            int curr=1;
            int j=i;
            while(j<n){
                curr*=nums[j];
                res=max(res,curr);
                j++;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i,k=0,0
        n=len(nums)
        if n==1 :
            return n
        res=[]
        while i<n-1:
            while i<n-1 and nums[i]==nums[i+1]:
                i+=1
            res.append(nums[i])
            i+=1
            k+=1
        if i==n-1 and nums[i-1]!=nums[i]:
            res.append(nums[i])
            k+=1
        for i in range(k):
            nums[i]=res[i]
        return k